src/Translations/TranslationExtractor.php

```diff
@@ -0,0 +1,145 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\PhpCs\Translations;
+
+use Becklyn\PhpCs\Translations\Integration\NameResolverIntegration;
+use Becklyn\PhpCs\Translations\Visitor\BackendTranslatorVisitor;
+use Becklyn\PhpCs\Translations\Visitor\CallbackValidationVisitor;
+use Becklyn\PhpCs\Translations\Visitor\ClassValidationVisitor;
+use Becklyn\PhpCs\Translations\Visitor\FormOptionLabelsVisitor;
+use Becklyn\PhpCs\Translations\Visitor\PropertyValidationVisitor;
+use Symfony\Component\Finder\Finder;
+use Translation\Extractor\Extractor;
+use Translation\Extractor\FileExtractor\FileExtractor;
+use Translation\Extractor\FileExtractor\PHPFileExtractor;
+use Translation\Extractor\FileExtractor\TwigFileExtractor;
+use Translation\Extractor\Model\SourceCollection;
+use Translation\Extractor\Model\SourceLocation;
+use Translation\Extractor\Visitor\Php\Symfony\ContainerAwareTrans;
+use Translation\Extractor\Visitor\Php\Symfony\ContainerAwareTransChoice;
+use Translation\Extractor\Visitor\Php\Symfony\FlashMessage;
+use Translation\Extractor\Visitor\Php\Symfony\FormTypeChoices;
+use Twig\Environment;
+use Twig\Loader\FilesystemLoader;
+
+class TranslationExtractor
+{
+    /**
+     * @param array $dirs
+     *
+     * @return string
+     */
+    public function extract (array $dirs) : array
+    {
+        if (empty($dirs))
+        {
+            return new SourceCollection();
+        }
+
+        $extractor = new Extractor();
+        $extractor->addFileExtractor($this->createPhpExtractor());
+        $extractor->addFileExtractor($this->createTwigExtractor($dirs));
+
+        $messages = [];
+        /** @var SourceLocation $location */
+        foreach ($extractor->extract($this->createFinder($dirs)) as $location)
+        {
+            $domain = $location->getContext()["domain"] ?? "messages";
```

Comment:
Do you happen to know whether the `TwigFileExtractor` also honors the `{% trans_default_domain "..." %}`? Otherwise we might get some false positives.

Reply:
Oops, the twig visitors were never registered.

Not sure whether it actually does support it, as the tag doesn't exist in Twig (but in Symfony) and therefor the extractor doesn't know it.